addons/account_analytic_default/models/product.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    rules_count = fields.Integer(compute='_compute_rules_count', string='# Analytic Rules')
+
+    @api.multi
+    def _compute_rules_count(self):
```

Comment:
No need @api.multi and also specify the correct field dependencies of a 'compute' method .